server/src/controllers/game.controller.js

```diff
@@ -201,8 +201,53 @@ class GameController {
   }
 
   // End game logic by checkmate or draw
-  // async endGame (req, res){
-  // }
+  async endGame(req, res, next) {
+    try {
+      const { game_id, user_id } = req.body;
+      const isGameExist = await GameRepo.fetchOne(game_id);
+
+      if (!isGameExist.data) {
+        return res
+          .status(400)
+          .send(response("Game does not exist", null, false));
+      }
+
+      if (!user_id) {
+        return res
+          .status(400)
+          .send(response("User does not exist", null, false));
+      }
+
+      let winner;
+      if (isGameExist.data.owner) {
```

Comment:
Your end game logic is wrong…

you are supposed to check if the incoming user_Id is that of owner or opponent. If it is for owner, you set is_owner_win (the one in the model) to true, else you set it to false.

we don’t have a checkmate key in our gameModel or design

Reply:
You also need to set the status to ended

Reply:
OK @NdubuisiJr